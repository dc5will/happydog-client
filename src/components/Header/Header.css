.Header {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    height: 60px;
    background-image: linear-gradient(90deg, rgb(192, 81, 81), rgb(180, 41, 41));
    /* background-color: rgb(192, 81, 81); */
}

.Header h1 {
    display: flex;
    justify-content: flex-start;
    align-items: center;
}

h1 {
    font-family: 'Dancing Script', cursive;
    font-size: 28px;
}

.Header a {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    color: white;
    text-decoration: none;
    margin: 10px;
}

.Header_not-logged-in, .Header_logged-in {
    display: flex;
    flex: 1;
    justify-content: flex-end;
    align-items: center;
    font-family: 'Montserrat',sans-serif;
    text-transform: uppercase;
    font-size: 14px;
}

.happyDogLogo {
    margin: 0px;
    margin-left: 5px;
}


@media only screen and (min-width: 600px) {
    .Header_not-logged-in, .Header_logged-in {
        display: flex;
        flex: 1;
        align-items: center;
        justify-content: flex-end;
        font-family: 'Montserrat',sans-serif;
        text-transform: uppercase;
        font-size: 20px;
        margin-right: 10px;
    }

    h1 {
        font-family: 'Dancing Script', cursive;
        font-size: 38px;
    }

    .Header {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;
        height: 85px;
    }
}

