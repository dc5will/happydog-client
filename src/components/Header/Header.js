import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./Header.css";

export default class Header extends Component {
  render() {
    return (
      <React.Fragment>
        <nav className="Header">
          <h1>
            <Link to="/">Happy dog</Link>
          </h1>
          <div className="Header_not-logged-in">
            <Link to="/login">Log in</Link>
            <Link to="/register">Register</Link>
          </div>
        </nav>
      </React.Fragment>
    );
  }
}
