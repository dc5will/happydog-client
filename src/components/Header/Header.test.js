import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("Header component", () => {
  it("renders without crashing", () => {
    shallow(<Header />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
