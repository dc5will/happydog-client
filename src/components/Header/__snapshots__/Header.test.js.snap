// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the UI as expected 1`] = `
<nav
  className="Header"
>
  <h1>
    <a
      href="/my-notes"
      onClick={[Function]}
    >
      Happy dog
    </a>
  </h1>
  <div
    className="Header_not-logged-in"
  >
    <a
      href="/login"
      onClick={[Function]}
    >
      Log in
    </a>
    <a
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</nav>
`;
